src/Cassandra.DataStax.Graph/Serialization/Tinkerpop/Structure/IO/GraphSON/PropertyDeserializer.cs

```diff
@@ -0,0 +1,40 @@
+﻿#region License
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#endregion
+
+using Cassandra.DataStax.Graph.Internal;
+using Gremlin.Net.Structure;
+using Newtonsoft.Json.Linq;
+
+namespace Cassandra.DataStax.Graph.Serialization.Tinkerpop.Structure.IO.GraphSON
+{
+    internal class PropertyDeserializer : IGraphSONDeserializer
```

Comment:
oh, we have to override core driver's custom deserializers to match TK behaviour, right?

Reply:
Yeah, to return the GLV classes instead of ours. The custom deserializers are only added on the `SubmitAsync` method (which is called by the GLV to send the requests). They aren't added if the user uses `session.ExecuteGraph()`.

Reply:
gotcha, thanks for the info!